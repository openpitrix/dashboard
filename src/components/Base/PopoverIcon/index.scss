@import '~scss/vars';

.targetLarge {
  width: 36px;
  height: 32px;
}
.targetNormal {
  width: 24px;
  height: 20px;
}
.target {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  border-radius: 2px;
  &.disable {
    opacity: 0.5;
  }
  &:hover {
    background: $N10;
  }
  &.visible {
    display: inline-flex;
    background: $N30;
    &:hover {
      background: $N30;
    }
  }
  .icon {
    position: static;
  }
}
.hoverTarget {
  &:hover {
    background: $N30;
  }
}

.container {
  display: inline-flex;
  align-items: center;
  &.visible {
    display: inline-flex !important;
  }
}
.showBorder {
  border: 1px solid $N30;
  border-radius: 2px;
  box-shadow: 0 1px 3px 0 rgba($N400, 0.07);

  .targetLarge {
    height: 30px;
  }
}

.popover {
  &[data-placement^='bottom'] {
    margin-left: 0;
    margin-top: 4px;
  }
}

.iconLarge {
  width: 20px;
  height: 20px;
}
.iconNormal {
  width: 16px;
  height: 16px;
}
