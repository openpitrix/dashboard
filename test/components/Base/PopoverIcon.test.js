import PopoverIcon from 'components/Base/PopoverIcon';

describe('Base/PopoverIcon', () => {
  it('basic render', () => {
    const wrapper = render(
      <PopoverIcon name={'view'} icon={'eye'}/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
