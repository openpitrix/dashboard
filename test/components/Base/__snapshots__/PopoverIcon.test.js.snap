// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base/PopoverIcon basic render 1`] = `
<div
  class="tooltip popover container"
>
  <div
    class="target target targetNormal"
  >
    view
    <span
      class="icon icon-eye icon targetNormal"
      style="width:16px;height:16px"
    >
      <svg
        class="qicon qicon-eye qicon-dark"
      >
        <use
          href="#qui-eye"
        />
      </svg>
    </span>
  </div>
  <div
    class="popper popover pi-popover-popper"
    style="position:absolute;pointer-events:none;opacity:0"
  >
    <div>
      <div
        class="pi-popover-content"
      />
    </div>
  </div>
</div>
`;
